scripts/build/git-id

```diff
@@ -1,12 +1,22 @@
 #!/bin/bash
 # get/save/load git-described version + SHA1 build id, usage: git-id [dir-to-save/read-ids]
 
+clean_dirty() {
+    if [ -n "$dirty" ]; then
+        files=$(git diff | grep ^diff | grep -v a/go.mod | grep -v a/go.sum)
+        if [ -z "$files" ]; then
+            dirty=""
+        fi
+    fi
+}
+
 unknown_version() {
     v=0.0.0
     if [ -d .git ]; then
         sha=$(git rev-parse --short HEAD)
         commits=$(git rev-list --count HEAD)
         dirty=$(git diff --quiet || echo "-dirty")
+        clean_dirty
```

Comment:
Wow, I didn't know that can be done with git diff. I'd love to use it... but that's no to enough to get rid of clean_dirty. AFAICT there is no similar mechanism for git describe --long --dirty.

Reply:
Sorry, I didn't get this- `clean_dirty()`only sets `dirty` to empty string. How is this related to `git describe --long --dirty?

Reply:
Well, you suggested removing clean_dirty altogether in favor of using equivalent built-in git mechanisms. There is no such thing for git-describe which is the primary way in the script for figuring out ids/version. But never mind, instead of removing it we can replace clean_dirty with exactly what you suggested...

Reply:
What I tried to suggest is to really drop `clean_dirty` and replace it with what I suggested. If you wan't to keep this one liner as a separate function I'd suggest to rename it something like:
```
is _dirty() {
  echo $(git diff --quiet -- ':!go.mod' ':!go.sum' || echo "-dirty")
}
```
and simplify the variable initialization inside the caller functions to just `dirty=""`

On a second thought, I somehow would like function name `get_dirty()` instead :smile:

Reply:
Yeah, you could have:
```
get_dirty() {
  echo $(git diff --quiet -- ':!go.mod' ':!go.sum' || echo "-dirty")
}

...

get_gitids () {
...
    dirty=$(get_dirty)
...
}
```
Right?